lighthouse/src/main.rs

```diff
@@ -659,8 +660,8 @@ fn run<E: EthSpec>(
                 executor.clone().spawn(
                     async move {
                         if let Err(e) = ProductionValidatorClient::new(context, config)
+                            .and_then(|mut vc| async move { vc.start_service().await })
```

Comment:
unfortunately this seems required to make the compiler happy, with:
```suggestion
                            .and_then(move |mut vc|  vc.start_service())
```
compiler gives:
```
       |
663    | ...                   .and_then(|mut vc| vc.start_service())
       |                                          ^^^^^^^^^^^^^^^^^^ returns a reference to data owned by the current function```